main_code/Email/inbox-dialogflow.py

```
@@ -0,0 +1,373 @@
+    expected_values_from_spreadsheet = ["Haltegriff Pl", "wils1170@msu.edu", "Gear", "GERTB125"]
+
+    action = get_action(example_tasks[3])
+    print(action)
+    take_action(action)
+
+
+def ModifyMessage(service, user_id, msg_id, msg_labels):
+  """Modify the Labels on the given Message.
+
+  Args:
+    service: Authorized Gmail API service instance.
+    user_id: User's email address. The special value "me"
+    can be used to indicate the authenticated user.
+    msg_id: The id of the message required.
+    msg_labels: The change in labels.
+
+  Returns:
+    Modified message, containing updated labelIds, id and threadId.
+  """
+  try:
+    message = service.users().messages().modify(userId=user_id, id=msg_id,
+                                                body=msg_labels).execute()
+
+    label_ids = message['labelIds']
+
+    print( 'Message ID: %s - With Label IDs %s' % (msg_id, label_ids))
+    return message
+  except errors.HttpError as error:
+    print ('An error occurred: %s' % error)
+
+
+def GetMessage(service, user_id, msg_id):
+  """Get a Message with given ID.
+
+  Args:
+    service: Authorized Gmail API service instance.
+    user_id: User's email address. The special value "me"
+    can be used to indicate the authenticated user.
+    msg_id: The ID of the Message required.
+
+  Returns:
+    A Message. the sender email
+  """
+  try:
+    message = service.users().messages().get(userId=user_id, id=msg_id).execute()
+
+    print( message['snippet'])
+
+
+
+    # attempt at sender email
+    emailline = str(message['payload']['headers'])
+    match = re.findall(r'[\w\.-]+@[\w\.-]+', emailline)
+    print("Message sender: " , match[1])
+    print("Message Id : ", message['id'])
+
+    return message['snippet'], match[1], message['id']
+  except errors.HttpError as error:
+    print ('An error occurred: %s' % error)

```

Again: make sure you really want to ignore errors and continue